lib/topological_inventory/scheduler/database_tools.rb

```diff
@@ -0,0 +1,15 @@
+require 'acts_as_tenant'
+
+module TopologicalInventory
+  module Scheduler
+    class DatabaseTools
+      REFRESH_STATES_TIME_THRESHOLD = 1.week.freeze
+
+      def self.remove_old_records
+        threshold = DateTime.now - REFRESH_STATES_TIME_THRESHOLD
+
+        rs_sum = RefreshState.where('updated_at <= ?', threshold).delete_all
```

Comment:
@pkomanek Did you check the cascade delete on `RefreshStatePart` will only nullify (doesn't delete) associated records? (`refresh_state_part_id foreign key =>  ON DELETE SET NULL`)